diff --git a/Lib/test/test_ssl.py b/Lib/test/test_ssl.py
index 54644e1596..64ac1744d1 100644
--- a/Lib/test/test_ssl.py
+++ b/Lib/test/test_ssl.py
@@ -220,6 +220,8 @@ class BasicSocketTests(unittest.TestCase):
         ssl.RAND_add(bytearray(b"this is a random bytearray object"), 75.0)
 
     @unittest.skipUnless(os.name == 'posix', 'requires posix')
+    @unittest.skipIf("test_ssl.test_random_fork" in os.environ["PYTHON_SKIP_TESTS"],
+                     'Fails on the advance-toolchain: https://github.com/advancetoolchain/advance-toolchain/issues/201')
     def test_random_fork(self):
         status = ssl.RAND_status()
         if not status:
