diff --git a/Lib/test/test_ssl.py b/Lib/test/test_ssl.py
index 54644e1596..67bedb7bd5 100644
--- a/Lib/test/test_ssl.py
+++ b/Lib/test/test_ssl.py
@@ -220,6 +220,7 @@ class BasicSocketTests(unittest.TestCase):
         ssl.RAND_add(bytearray(b"this is a random bytearray object"), 75.0)
 
     @unittest.skipUnless(os.name == 'posix', 'requires posix')
+    @unittest.skip('Fails on the advance-toolchain: https://github.com/advancetoolchain/advance-toolchain/issues/201')
     def test_random_fork(self):
         status = ssl.RAND_status()
         if not status:
