diff --git a/expat/configure.ac b/expat/configure.ac
index c24f518..6c1369f 100644
--- a/expat/configure.ac
+++ b/expat/configure.ac
@@ -237,7 +237,7 @@ AS_HELP_STRING([--without-docbook],
 
 AC_ARG_VAR([DOCBOOK_TO_MAN], [docbook2x-man command])
 AS_IF([test "x$with_docbook" != xno],
-  [AC_CHECK_PROGS([DOCBOOK_TO_MAN], [docbook2x-man docbook2man docbook-to-man],
+  [AC_CHECK_PROGS([DOCBOOK_TO_MAN], [db2x_docbook2man docbook2x-man docbook2man docbook-to-man],
   [if test "x$with_docbook" != xcheck; then 
     AC_MSG_ERROR([Required program 'docbook2x-man' not found.])])])
 
