diff --git a/libnxz.h b/libnxz.h
index f635588..f2f3316 100644
--- a/libnxz.h
+++ b/libnxz.h
@@ -70,7 +70,7 @@ extern int nx_deflateInit2_(void *strm, int level, int method, int windowBits,
 extern int nx_deflate(void *strm, int flush);
 extern int nx_deflateEnd(void *strm);
 extern ulong nx_deflateBound(void *strm, ulong sourceLen);
-extern int nx_deflateParams(z_streamp strm, int level, int strategy);
+extern int nx_deflateParams(void *strm, int level, int strategy);
 extern int nx_deflateSetHeader(void *strm, void *head);
 extern int nx_deflateCopy(void *dest, void *source);
 extern int nx_deflateReset(void *strm);
